### PR TITLE
EPMDJ-1020. Implemented an admin ws topic for getting agent's plugin config

### PR DESCRIPTION
**Pull request by s0larm00n**
Implemented an admin ws topic for getting agent's plugin config instead of a rest endpoint.
Fixed the UpdatePlugin endpoint to recieve a raw json representation of the config.

### CODE REVIEW

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/ServerWsTopics.kt [hunk 1]
```diff
@@ -89,6 +89,10 @@ class ServerWsTopics(override val kodein: Kodein) : KodeinAware {
                         pluginWebSocket
                     }
                 }
+
+                topic<WsRoutes.GetPluginConfig> { ids, _ ->
+                    agentManager[ids.agentId]?.plugins?.find { it.id == ids.pluginId }?.config
```
Comment (IgorKey) [inline drill-admin/src/main/kotlin/com/epam/drill/endpoints/ServerWsTopics.kt]: ids? what is it?  agentId, pluginId??? )))

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/agent/DrillServerWs.kt [hunk 2]
```diff
@@ -56,38 +56,37 @@ class DrillServerWs(override val kodein: Kodein) : KodeinAware {
...
-                    sessionStorage.sendTo(
-                        Message(
-                            MessageType.MESSAGE,
-                            event.destination,
-                            serializer stringify resolve
-                        )
-                    )
+                val resolve = resolve(destination, sessionStorage)!!
+                val message = when (resolve) {
+                    is Collection<*> -> Json.stringify(resolve)
```
Comment (IgorKey) [inline drill-admin/src/main/kotlin/com/epam/drill/endpoints/agent/DrillServerWs.kt]: expand this!
try to rewrite this part of bf to **when** operator.
when( List) -> serializer.list
when(Set) -> serializer.set
etc..
Remove autogen Json

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/ServerWsTopics.kt (hunk 1 above)
Comment (s0larm00n) [reply drill-admin/src/main/kotlin/com/epam/drill/endpoints/ServerWsTopics.kt]: this is for agent's and plugin's ids

Comment (s0larm00n) [reply drill-admin/src/main/kotlin/com/epam/drill/endpoints/ServerWsTopics.kt]: is it better to call it some 'x', like at row 68?

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/ServerWsTopics.kt [hunk 3]
```diff
@@ -89,6 +89,10 @@ class ServerWsTopics(override val kodein: Kodein) : KodeinAware {
                         pluginWebSocket
                     }
                 }
+
+                topic<WsRoutes.GetPluginConfig> { id, _ ->
+                    agentManager[id.agent]?.plugins?.find { it.id == id.plugin }?.config
```
Comment (IgorKey) [inline drill-admin/src/main/kotlin/com/epam/drill/endpoints/ServerWsTopics.kt]: ID....
Oh

File: drill-admin/src/main/kotlin/com/epam/drill/endpoints/WsTopic.kt [hunk 4]
```diff
@@ -75,3 +68,31 @@ class Temp<T, R>(val block: (R, MutableSet<DrillWsSession>) -> T) {
     }
 }
 
+@UseExperimental(ImplicitReflectionSerializer::class)
+fun serialize(value: Any?): String {
+    if (value == null) return ""
+    val serializer = when (value) {
+        is String -> null
```
Comment (IgorKey) [inline drill-admin/src/main/kotlin/com/epam/drill/endpoints/WsTopic.kt:75]: may be StringSerializer

Comment (s0larm00n) [reply drill-admin/src/main/kotlin/com/epam/drill/endpoints/WsTopic.kt]: no, may not. it turns the origin string to total mess

**Review by IgorKey — APPROVED** [review]

**Pull request merged**